pkg/kube/controllers/statefulset/statefulset_rollout_util.go

```diff
@@ -0,0 +1,120 @@
+package statefulset
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"time"
+
+	"code.cloudfoundry.org/cf-operator/pkg/kube/util"
+
+	"code.cloudfoundry.org/quarks-utils/pkg/pointers"
+
+	"code.cloudfoundry.org/cf-operator/pkg/bosh/manifest"
+	"code.cloudfoundry.org/quarks-utils/pkg/ctxlog"
+	podutil "code.cloudfoundry.org/quarks-utils/pkg/pod"
+	"k8s.io/api/apps/v1beta2"
+	corev1 "k8s.io/api/core/v1"
+	crc "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ConfigureStatefulSetForRollout configures a stateful set for canarying and rollout
+func ConfigureStatefulSetForRollout(statefulSet *v1beta2.StatefulSet) {
+	statefulSet.Spec.UpdateStrategy.Type = v1beta2.RollingUpdateStatefulSetStrategyType
+	//the canary rollout is for now directly started, the might move to a webhook instead
+	statefulSet.Spec.UpdateStrategy.RollingUpdate = &v1beta2.RollingUpdateStatefulSetStrategy{
+		Partition: pointers.Int32(util.MaxInt32(util.MinInt32(*statefulSet.Spec.Replicas-1, statefulSet.Status.Replicas), 0)),
+	}
+	statefulSet.Annotations[AnnotationCanaryRollout] = rolloutStatePending
```

Comment:
should not store state here - state should be set on the qsts

Reply:
see above